src/integration/sink/tls_test.go

```diff
@@ -0,0 +1,76 @@
+package sink_test
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/newrelic/nri-kubernetes/v3/internal/config"
+	"github.com/newrelic/nri-kubernetes/v3/src/integration/sink"
+)
+
+func testServer() (*httptest.Server, error) {
+	server := httptest.NewUnstartedServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		_, _ = io.Copy(io.Discard, r.Body)
+		rw.WriteHeader(http.StatusOK)
+	}))
+
+	serverCert, err := tls.LoadX509KeyPair("testdata/server+3.pem", "testdata/server+3-key.pem")
+	if err != nil {
+		return nil, fmt.Errorf("loading server certificate: %w", err)
+	}
+
+	caCert, err := os.ReadFile("testdata/rootCA.pem")
+	if err != nil {
+		return nil, fmt.Errorf("loading CA certificate: %w", err)
+	}
+	caPool := x509.NewCertPool()
+	caPool.AppendCertsFromPEM(caCert)
+
+	tlsConfig := tls.Config{
+		Certificates: []tls.Certificate{serverCert},
+		ClientCAs:    caPool,
+	}
+
+	server.TLS = &tlsConfig
+	server.StartTLS()
+
+	return server, nil
+}
+
+func TestTlsClient(t *testing.T) {
+	server, err := testServer()
```

Comment:
For test helpers it maybe useful to include `*testing.T` as a parameter and drop the returning error, if it actually fails it a bit easier to find where it failed. Besides, in case the helper is widely used, you avoid checking the error each time it is used.

Something like this:
```go
func testServer(t *testing.T) *httptest.Server {
    // ...
    serverCert, err := tls.LoadX509KeyPair("testdata/server+3.pem", "testdata/server+3-key.pem")
    if err != nil {
         return t.Fatalf("error loading server certificate: %w", err)
    }
    // ...
}

// ...
server := testServer(t)
```

Reply:
lgtm, I gave that a try on https://github.com/newrelic/nri-kubernetes/pull/338/commits/e9faf94dcb3bdad67e9cdd7e5aeadd6578f0a7d6